src/Bicep.Types/Concrete/FunctionType.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Azure.Bicep.Types.Concrete;
+
+public class FunctionType : TypeBase
+{
+    [JsonConstructor]
+    public FunctionType(string name, ITypeReference output, IReadOnlyList<FunctionTypeParameter> parameters, string? description)
+        => (Name, Output, Parameters, Description) = (name, output, parameters, description);
+
+    public string Name { get; }
+
+    public ITypeReference Output { get; }
+
+    public IReadOnlyList<FunctionTypeParameter> Parameters { get; }
+
+    public string? Description { get; }
+}
+
+public class FunctionTypeParameter
```

Comment:
nit: is this a *type parameter of a function*? or the *type of a parameter of a function*?

eg: `Func<T>` vs `Func(T)`

Reply:
The latter. Any suggestions for naming?

Reply:
`FunctionParameterType`

Reply:
Renamed to `FunctionParameter`. I avoided `FunctionParameterType` because it isn't a type.